autopush/senderids.py

```diff
@@ -40,63 +41,73 @@
 
 class SenderIDs(object):
     """Handle Read, Write and cache of SenderID values from S3"""
-    _updated = 0
     _expry = SENDERID_EXPRY
     _senderIDs = {}
     _use_s3 = True
     KEYNAME = "senderids"
+    service = None
 
     def __init__(self, args):
         """Optionally load or fetch the set of SenderIDs from S3"""
         self.conn = S3Connection()
-        self.ID = args.get("s3_bucket", DEFAULT_BUCKET)
+        self.ID = args.get("s3_bucket", DEFAULT_BUCKET).lower()
         self._expry = args.get("senderid_expry", SENDERID_EXPRY)
         self._use_s3 = args.get("use_s3", True)
         senderIDs = args.get("senderid_list", {})
+        self.service = None
         if senderIDs:
             if type(senderIDs) is not dict:
                 log.err("senderid_list is not a dict. Ignoring")
             else:
+                # We're initializing, so it's ok to block.
                 self.update(senderIDs)
-        self._refresh()
 
-    def _write(self, bucket, senderIDs):
+    def start(self):
+        if self._use_s3:
+            log.msg("Starting SenderID service...")
+            self.service = TimerService(self._expry, self._refresh)
+            self.service.startService()
+
+    def _write(self, senderIDs, *args):
         """Write a list of SenderIDs to S3"""
+        bucket = self.conn.get_bucket(self.ID)
         key = Key(bucket)
         key.key = self.KEYNAME
         key.set_contents_from_string(json.dumps(senderIDs))
         self._senderIDs = senderIDs
-        self._updated = time.time()
 
-    def _create(self, senderIDs):
+    def _err(self, state):
+        if (isinstance(state.value, S3ResponseError) and
+                state.value.reason == 'Not Found'):
+            self._create()
```

Comment:
Should this be `deferToThread(self._create)`, or is this guaranteed to only happen at startup?